src/sentry/tasks/backfill_outboxes.py

```diff
@@ -36,17 +36,24 @@ def get_backfill_key(table_name: str) -> str:
 
 
 def get_processing_state(table_name: str) -> Tuple[int, int]:
+    result: Tuple[int, int]
     with redis.clusters.get("default").get_local_client_for_key("backfill_outboxes") as client:
         key = get_backfill_key(table_name)
         v = client.get(key)
         if v is None:
             result = (0, 1)
             client.set(key, json.dumps(result))
-            return result
-        lower, version = json.loads(v)
-        if not (isinstance(lower, int) and isinstance(version, int)):
-            raise TypeError("Expected processing data to be a tuple of (int, int)")
-        return lower, version
+        else:
+            lower, version = json.loads(v)
+            if not (isinstance(lower, int) and isinstance(version, int)):
+                raise TypeError("Expected processing data to be a tuple of (int, int)")
+            result = lower, version
+        metrics.gauge(
+            "backfill_outboxes.low_bound",
+            result[1],
+            tags=dict(table_name=table_name, version=result[0]),
+        )
```

Comment:
You will likely want `sample_rate=1.0` on these metrics as we have a 10% sample rate in saas.

Reply:
🙏